File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int first_occur(vector<int> &nums,int target){
        int i=0;
        int j=nums.size()-1;
        int index=-1;
        while (i<=j){
            int mid=i+(j-i)/2;
            if(nums[mid]>=target){
                j=mid-1;
            }
            else{
                i=mid+1;
            }
            if (nums[mid]==target){
                index=mid;
            }
            
        }
        return index;
    }
    
    int last_occur(vector<int> & nums,int target){
        int i=0;
        int j=nums.size()-1;
        int index=-1;
        while(i<=j){
            int mid=i+(j-i)/2;
            if(nums[mid]<=target){
                i=mid+1;
            }
            else{
                j=mid-1;
            }
            if(nums[mid]==target){
                index=mid;
            }
        }
        return index;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
        
        vector<int> ans;
        ans.push_back(first_occur(nums,target));
        ans.push_back(last_occur(nums,target));
        return ans;
        
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def last(nums,target):
            index1=0
            low=0
            high=len(nums)-1
            while(low<=high):
                mid=low+(high-low)//2
                if nums[mid]<=target:
                    low=mid+1
                    index1=mid
                    
                    
                else:
                    high=mid-1
            if len(nums)==0 or nums[index1]!=target:
                return -1
            else:
                return index1
        def first(nums,target):
            
            low=0
            index=0
            high=len(nums)-1
            while(low<high):
                mid=low+(high-low)//2
                if nums[mid]>=target:
                    high=mid
                else:
                    low=mid+1
                    index=low
            if len(nums)==0 or nums[index]!=target :
                return -1
            else:
                return index
        a=[-1,-1]    
        a[0]=first(nums,target)
        a[1]=last(nums,target)
        return a
        
        
            
        